zerver/tests/test_realm_emoji.py

```diff
@@ -79,6 +79,30 @@ def test_upload(self) -> None:
         author = UserProfile.objects.get(id=test_emoji["author_id"])
         self.assertEqual(author.email, email)
 
+    def test_override_built_in_emoji_by_admin(self) -> None:
+        # Tests for an event where an admin successfully overrides the built-in emoji.
+        user = self.example_user("iago")
+        email = user.email
+        self.login_user(user)
+        with get_test_image_file("img.png") as fp1:
+            emoji_data = {"f1": fp1}
+            result = self.client_post("/json/realm/emoji/smile", info=emoji_data)
+        self.assert_json_success(result)
+        self.assertEqual(200, result.status_code)
+        realm_emoji = RealmEmoji.objects.get(name="smile")
+        self.assertEqual(realm_emoji.author.email, email)
+
+    def test_override_built_in_emoji_by_non_admin(self) -> None:
+        # Tests when a non admin tries to override the built-in emoji.
+        self.login("othello")
+        with get_test_image_file("img.png") as fp1:
+            emoji_data = {"f1": fp1}
+            result = self.client_post("/json/realm/emoji/laughing", info=emoji_data)
+        self.assert_json_error(
+            result,
+            "An emoji with this name already exists. Only administrators can override built-in emoji.",
+        )
```

Comment:
See our testing philosophy page, but we should do both the failure and success tasks in a single test.